test/groovy/vars/OdsComponentStageRolloutOpenShiftDeploymentSpec.groovy

```diff
@@ -28,17 +28,18 @@ class OdsComponentStageRolloutOpenShiftDeploymentSpec extends PipelineSpockTestB
       componentId: 'bar'
   ]
 
-  def "run successfully without Tailor"() {
+  def "run successfully without Tailor [DeploymentConfig]"() {
     given:
     def c = config + [environment: 'dev', targetProject: 'foo-dev', openshiftRolloutTimeoutRetries: 6]
     IContext context = new Context(null, c, logger)
     OpenShiftService openShiftService = Stub(OpenShiftService.class)
-    openShiftService.resourceExists(*_) >> true
-    openShiftService.getLatestVersion(*_) >> 123
-    openShiftService.rollout(*_) >> "${config.componentId}-124"
+    openShiftService.resourceExists('foo-dev', 'DeploymentConfig', 'bar') >> true
+    openShiftService.resourceExists('foo-dev', 'Deployment', 'bar') >> false
+    openShiftService.getRevision('foo-dev', 'DeploymentConfig', 'bar') >> 123
+    openShiftService.rollout('foo-dev', 'DeploymentConfig', 'bar', 123, 5) >> "bar-124"
     // test the handover of the poddata retries
-    openShiftService.getPodDataForDeployment(_, _, 6) >> [ deploymentId: "${config.componentId}-124" ]
-    openShiftService.getImagesOfDeploymentConfig (*_) >> [[ repository: 'foo', name: 'bar' ]]
+    openShiftService.getPodDataForDeployment('foo-dev', 'DeploymentConfig', 'bar-124', 6) >> [[ deploymentId: "bar-124" ]]
```

Comment:
Out of curiosity - why has it changed to a double array?

Reply:
Note that this is actually not a double array, but `List<Map>`, whereas previously it was just `Map`. 

The reason this is a list now is because one deployment can have multiple pods ("replicas"). I would consider this a pretty typical use case, which we should support eventually in the docs. The issue tracking this is #390. In this PR, I'm already dealing with multiple pods properly from a technical perspective. Previously, we would only look at the first pod, not all of them, which is not fully correct. Now we consider all pods when gathering the data, but the orchestration pipeline then only uses the first item in the list to put it into the docs (and throws away information about any further pods).

Reply:
Interesting! Thanks for the explanation Michael